src/metabase/data_apps/models.clj

```diff
@@ -0,0 +1,121 @@
+(ns metabase.data-apps.models
+  (:require
+   [metabase.models.interface :as mi]
+   [metabase.util.malli :as mu]
+   [metabase.util.malli.registry :as mr]
+   [methodical.core :as methodical]
+   [toucan2.core :as t2]))
+
+(set! *warn-on-reflection* true)
+
+(methodical/defmethod t2/table-name :model/DataApp           [_model] :data_app)
+(methodical/defmethod t2/table-name :model/DataAppDefinition [_model] :data_app_definition)
+(methodical/defmethod t2/table-name :model/DataAppRelease    [_model] :data_app_release)
+
+(doseq [model [:model/DataApp
+               :model/DataAppDefinition
+               :model/DataAppRelease]]
+  (derive model :metabase/model))
+
+(derive :model/DataApp :hook/timestamped?)
+(derive :model/DataApp :hook/entity-id)
+(derive :model/DataAppDefinition :hook/created-at-timestamped?)
+
```

Comment:
We should probably add `updated-at` to `DataAppRelease` so we can see when it was retracted 🤔 

Or... we could add a `retracted_at` timestamp, and use that being non-nil as the flag 🤔 

Probably renaming `released_at` to `created_at` is also fine, unless we want to support un-retracting, which I don't think we do. We can instead allow a version to get re-released as a new row.